corehq/apps/mobile_auth/models.py

```diff
@@ -44,3 +47,43 @@ def key_for_time(cls, domain, user_id, now):
             include_docs=True,
         ).first()
         return key_record
+
+    @classmethod
+    def _migration_get_sql_model_class(cls):
+        return SQLMobileAuthKeyRecord
+
+    @classmethod
+    def _migration_get_fields(cls):
+        return ["domain", "user_id", "valid", "expires", "type", "key"]
+
+
+def _default_uuid():
+    return uuid.uuid4().hex
+
+
+def _default_key():
+    return generate_aes_key().decode('utf-8')
+
+
+class SQLMobileAuthKeyRecord(SyncSQLToCouchMixin, models.Model):
+    uuid = models.UUIDField(primary_key=True, db_index=True, default=_default_uuid, db_column='id')
```

Comment:
Sorry for not noticing this earlier. `primary_key=True` implicitly creates and index, so `db_index=True` is unnecessary and possibly even harmful (if it creates an extra/unnecessary index).

From the docs:

[primary_key](https://docs.djangoproject.com/en/3.0/ref/models/fields/#primary-key)
> `primary_key=True` implies `null=False` and `unique=True`.

[unique](https://docs.djangoproject.com/en/3.0/ref/models/fields/#unique)
> Note that when `unique` is `True`, you don’t need to specify `db_index`, because `unique` implies the creation of an index.


```suggestion
    uuid = models.UUIDField(primary_key=True, default=_default_uuid, db_column='id')
```

Reply:
I didn't realize the old `uuid` field on the Couch model was defined as a read-only property:

https://github.com/dimagi/commcare-hq/blob/0c6699ac83a018b93ca09baf902b1822ce10bc7f/corehq/apps/mobile_auth/models.py#L35-L37

I know we agreed on the compromise of `db_column='id'`, but I think it would be even better to do what was done on the Couch model. I don't know why I didn't think of this earlier.
```py
class SQLMobileAuthKeyRecord(SyncSQLToCouchMixin, models.Model):
    id = models.UUIDField(primary_key=True, default=_default_uuid)
    ...

    @property
    def uuid(self):
        return self.id
```
I'm not asking you to change it, just pointing that out for future situations.

Reply:
Updated in 1c922b81d404bbb35afd4947ed42c850fe99fda0 - I think we're now both happy with this code, but feel free to ping me if not!